rs/embedders/tests/spec_tests.rs

```diff
@@ -20,21 +20,37 @@ const FILES_TO_SKIP: &[&str] = &["names.wast"];
 mod convert {
     use wasmtime::Store;
     use wast::{
-        core::{HeapType, NanPattern, V128Pattern, WastArgCore},
-        token::{Float32, Float64},
+        core::{AbstractHeapType, HeapType, NanPattern, V128Pattern, WastArgCore},
+        token::{F32, F64},
         WastArg, WastRet,
     };
 
     fn heap_type(heap_type: HeapType) -> wasmtime::Val {
         match heap_type {
-            HeapType::Func => wasmtime::Val::FuncRef(None),
-            HeapType::Extern => wasmtime::Val::ExternRef(None),
-            HeapType::Any
-            | HeapType::Eq
-            | HeapType::Array
-            | HeapType::I31
-            | HeapType::Index(_)
-            | HeapType::Struct => {
+            HeapType::Abstract { shared: _, ty } => match ty {
+                AbstractHeapType::Func => wasmtime::Val::FuncRef(None),
+                AbstractHeapType::Extern => wasmtime::Val::ExternRef(None),
+                AbstractHeapType::Any
+                | AbstractHeapType::Eq
+                | AbstractHeapType::Struct
+                | AbstractHeapType::Array
+                | AbstractHeapType::I31
+                | AbstractHeapType::NoFunc
+                | AbstractHeapType::NoExtern
+                | AbstractHeapType::None => {
+                    panic!(
+                        "Unable to handle heap type {:?}. The GC proposal isn't supported",
+                        heap_type
+                    )
```

Comment:
Is this error message correct? We use the same below for `HeapType::Concrete(_)`.